2025/4/20/cancellable-timeout/cancellable-timeout.ts

```
@@ -0,0 +1,15 @@
+  (...args: Array<any>): void;
+}
+
+export default function cancellableTimeout(
+  callback: TimeoutCallback,
+  delay?: number,
+  ...args: Array<any>
+): () => void {
+  const timeoutId = setTimeout(callback, delay, ...args);
+
+  return function (): void {
+    return clearTimeout(timeoutId);
```

Biome (1.9.4)

[error] The function should not return a value because its return type is void.

The function is here:

'void' signals the absence of value. The returned value is likely to be ignored by the caller.

(lint/correctness/noVoidTypeReturn)